### PR TITLE
Refactor request handling

### PR DESCRIPTION
**Pull request by Johannes-Schneider**

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java [hunk 1]
```diff
@@ -53,227 +52,203 @@
...
-            } else {
-                response.status(HttpStatusCodes.ERROR);
-                return blockingEventTypes;
-            }
         }
-
-        response.body("");
-        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
-        return response.body();
+		if (!eventType.isDeletable()) {
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java:152]: BLOCKER NullPointerException might be thrown as 'eventType' is nullable here [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![BLOCKER][BLOCKER] 1 blocker

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'

File: src/test/java/de/hpi/bpt/argos/api/EventTypeEndpointTest.java [hunk 2]
```diff
@@ -106,18 +106,18 @@ public void testCreateEventType() throws Exception {
...
     @Test
-    public void testCreateEventType_EmptyJson_BadRequest() {
+    public void testCreateEventType_EmptyJson_Error() {
         RestRequest request = RestRequestFactoryImpl.getInstance()
                 .createPostRequest(ARGOS_REST_HOST, EventTypeEndpoint.getCreateEventTypeBaseUri());
 
         JsonObject emptyJson = new JsonObject();
         request.setContent(serializer.toJson(emptyJson));
 
-        assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
```
Comment (frederike-ramin) [inline src/test/java/de/hpi/bpt/argos/api/EventTypeEndpointTest.java:116]: To me it's not a good idea to change this behavior. maybe just catch the parser exception and throw a halt?

**Review by frederike-ramin — APPROVED** [review]
Nice refactoring!

File: src/test/java/de/hpi/bpt/argos/api/EventTypeEndpointTest.java (hunk 2 above)
Comment (Johannes-Schneider) [reply src/test/java/de/hpi/bpt/argos/api/EventTypeEndpointTest.java:116]: I tried... But the code will lose its beauty when catching all the json NPEs.

File: src/main/java/de/hpi/bpt/argos/util/RestEndpointUtilImpl.java [hunk 3]
```diff
@@ -112,20 +112,25 @@ public String executeRequest(Logger logger, Request request, Response response,
 		try {
 			response.body((String) route.handle(request, response));
 		} catch (HaltException e) {
-			logSendingResponse(logger, request, e.statusCode(), e.body());
-			throw e;
```
Comment (frederike-ramin) [inline src/main/java/de/hpi/bpt/argos/util/RestEndpointUtilImpl.java]: You don't want to rethrow that?

**Review by frederike-ramin — CHANGES_REQUESTED** [review]

File: src/main/java/de/hpi/bpt/argos/util/RestEndpointUtilImpl.java (hunk 3 above)
Comment (Johannes-Schneider) [reply src/main/java/de/hpi/bpt/argos/util/RestEndpointUtilImpl.java]: Actually no. This way I can handle all responses myself and dont have to hand them back to the framework. (This becomes important, when setting the response type in error-cases)

**Review by frederike-ramin — APPROVED** [review]

**Pull request merged**